gocache.go

```diff
@@ -274,7 +274,9 @@ func (s *Server) handleGet(ctx context.Context, req *progRequest) (pr *progRespo
 	// Safety check: The output ID should be hex-encoded and non-empty.
 	if outputID == "" {
 		return nil, errors.New("get: empty output ID")
-	} else if _, err := hex.DecodeString(outputID); err != nil {
+	}
+	outputHexID, err := hex.DecodeString(outputID)
```

Comment:
It's a somewhat minor thing, but the name here is a bit confusing: What we're calling `outputHexID` is _not_ the hex version, it's the raw string. We do want the raw string, so perhaps `rawOutputID` would be better. 

Or, though it would be a bigger change, maybe swap `outputID` and `outputHexID` throughout this function.

Reply:
On second thought, I can just handle that as a follow-up.